src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -706,20 +702,56 @@ private static ISymbol GetEventAccessor(IEventSymbol iEvent, IOperation operatio
                     {
                         if (IsNotSuppressedByCallSite(operationAttributes, callSiteAttributes, msBuildPlatforms, out var notSuppressedAttributes))
                         {
-                            platformSpecificOperations.TryAdd(operation, notSuppressedAttributes);
+                            platformSpecificOperations.TryAdd(new KeyValuePair<IOperation, ISymbol>(operation, symbol), notSuppressedAttributes);
+                        }
+                    }
+                    else
+                    {
+                        if (TryCopyAttributesNotSuppressedByMsBuild(operationAttributes, msBuildPlatforms, out var copiedAttributes))
+                        {
+                            platformSpecificOperations.TryAdd(new KeyValuePair<IOperation, ISymbol>(operation, symbol), copiedAttributes);
+                        }
+                    }
+                }
+            }
+
+            static void CheckImmedaiteAttributes(IOperation operation, OperationAnalysisContext context, PooledConcurrentDictionary<KeyValuePair<IOperation, ISymbol>, SmallDictionary<string, PlatformAttributes>> platformSpecificOperations,
```

Comment:
```suggestion
            static void CheckImmediateAttributes(IOperation operation, OperationAnalysisContext context, PooledConcurrentDictionary<KeyValuePair<IOperation, ISymbol>, SmallDictionary<string, PlatformAttributes>> platformSpecificOperations,
```

Reply:
It seems like the new added method is the exact clone of `CheckOperationAttributes`. Instead of duplicate implementations, can we just past in a new flag to `CheckOperationAttributes`, say `bool onlyImmediateSymbol`, which is then passed down to `TryGetOrCreatePlatformAttributes` to perform the appropriate logic?

Reply:
I have actually moved the duplicate section into a separate method